### PR TITLE
Allow setting cloud account for non-production environments

### PR DESCRIPTION
**Pull request by mpolden**
@jonmv

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java [hunk 1]
```diff
@@ -702,7 +702,7 @@ public void deploy(ApplicationId id, JobType type, Optional<Version> platform, A
         controller.applications().lockApplicationOrThrow(TenantAndApplicationId.from(id), application -> {
             if ( ! application.get().instances().containsKey(id.instance()))
                 application = controller.applications().withNewInstance(application, id);
-
+            controller.applications().validatePackage(applicationPackage, application.get());
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/JobController.java:705]: 👍

**Review by jonmv — APPROVED** [review]
Nice!

**Comment by mpolden** [comment]
`com.yahoo.vespa.clustercontroller.core.MasterElectionTest` strikes again 😞

**Comment by mpolden** [comment]
Tried running a full build locally, but then `FleetControllerTest.waitForStableSystem` failed. 😆

**Pull request merged**